### ISSUE TITLE
ELI5 no longer available, but referenced in example code

### ISSUE THREAD

**Issue by drdsgvo** [huggingface/datasets#6605]
Here, an example code is given:
https://huggingface.co/docs/transformers/tasks/language_modeling

This code + article references the ELI5 dataset.
ELI5 is no longer available, as the ELI5 dataset page states: https://huggingface.co/datasets/eli5
"Defunct: Dataset "eli5" is defunct and no longer accessible due to unavailability of the source data.
Reddit recently [changed the terms of access](https://www.reddit.com/r/reddit/comments/12qwagm/an_update_regarding_reddits_api/) to its API, making the source data for this dataset unavailable.
"

Please change the example code to use a different dataset.

**Comment by mariosasko** [comment]
Addressed in https://github.com/huggingface/transformers/pull/28715.
